velox/dwio/dwrf/common/FileMetadata.h

```diff
@@ -828,6 +891,973 @@ class FooterWrapper : public ProtoWrapperBase {
+    return format_ == DwrfFormat::kDwrf
+        ? dwrfPtr()->set_headerlength(headerLength)
+        : orcPtr()->set_headerlength(headerLength);
+  }
+
+  void setContentLength(uint64_t contentLength) const {
+    return format_ == DwrfFormat::kDwrf
+        ? dwrfPtr()->set_contentlength(contentLength)
+        : orcPtr()->set_contentlength(contentLength);
+  }
+
+  void setRowIndexStride(uint32_t rowIndexStride) const {
+    format_ == DwrfFormat::kDwrf ? dwrfPtr()->set_rowindexstride(rowIndexStride)
+                                 : orcPtr()->set_rowindexstride(rowIndexStride);
+  }
+
+  void setNumberOfRows(uint64_t numberOfRows) const {
+    format_ == DwrfFormat::kDwrf ? dwrfPtr()->set_numberofrows(numberOfRows)
+                                 : orcPtr()->set_numberofrows(numberOfRows);
+  }
+
+  void setRawDataSize(uint64_t numberOfRows) const {
+    if (format_ == DwrfFormat::kDwrf) {
+      dwrfPtr()->set_rawdatasize(numberOfRows);
+    }
+  }
+
+  void setWriter(uint32_t writer) const {
+    if (format_ == DwrfFormat::kOrc) {
+      orcPtr()->set_writer(writer);
+    }
+  }
+
+  void setCheckSumAlgorithm(proto::ChecksumAlgorithm checksum) const {
+    if (format_ == DwrfFormat::kDwrf) {
+      dwrfPtr()->set_checksumalgorithm(checksum);
+    }
+  }
+
+  void addStripeCacheOffsets(uint32_t stripeCacheOffsets) const {
+    if (format_ == DwrfFormat::kDwrf) {
+      dwrfPtr()->add_stripecacheoffsets(stripeCacheOffsets);
+    } else {
+      //
+    }
+  }
+
+  TypeWriteWrapper addTypes() const {
+    return format_ == DwrfFormat::kDwrf
+        ? TypeWriteWrapper(dwrfPtr()->add_types())
+        : TypeWriteWrapper(orcPtr()->add_types());
+  }
+
+  UserMetadataItemWriteWrapper addMetadata() const {
+    return format_ == DwrfFormat::kDwrf
+        ? UserMetadataItemWriteWrapper(dwrfPtr()->add_metadata())
+        : UserMetadataItemWriteWrapper(orcPtr()->add_metadata());
+  }
+
+  ColumnStatisticsWriteWrapper addStatistics() const {
+    return format_ == DwrfFormat::kDwrf
+        ? ColumnStatisticsWriteWrapper(dwrfPtr()->add_statistics())
+        : ColumnStatisticsWriteWrapper(orcPtr()->add_statistics());
+  }
+
+  int typesSize() const {
+    return format_ == DwrfFormat::kDwrf ? dwrfPtr()->types_size()
+                                        : orcPtr()->types_size();
+  }
+
+  int statisticsSize() const {
+    return format_ == DwrfFormat::kDwrf ? dwrfPtr()->statistics_size()
+                                        : orcPtr()->statistics_size();
+  }
+
+  uint64_t contentLength() const {
+    return format_ == DwrfFormat::kDwrf ? dwrfPtr()->contentlength()
+                                        : orcPtr()->contentlength();
+  }
+
+  inline int stripesSize() const {
+    VELOX_CHECK_EQ(format_, DwrfFormat::kDwrf);
+    return dwrfPtr()->stripes_size();
+  }
+
+  inline proto::Encryption* mutableEncryption() {
+    VELOX_CHECK_EQ(format_, DwrfFormat::kDwrf);
+    return dwrfPtr()->mutable_encryption();
+  }
+
```

Comment:
Can we add `numberOfRows()` here?  Some application is calling this

Reply:
@Yuhta Already added, see lines 1423-1426. Thanks.

Reply:
Can you add the following as well?

FooterWriteWrapper:
* statistics_size
* rawdatasize
* add_statistics
* statistics

StripeInformation:
* set_offset
* set_indexlength
* set_datalength
* set_footerlength
* set_groupsize

There could be more as our engineer trying to build with the change

Reply:
I have added all the above methods, as follows:

FooterWriteWrapper:
- statistics_size  -> statisticsSize
- rawdatasize  ->  rawDataSize
- add_statistics  -> addStatistics
- statistics  ->  statistics

StripeInformation:

- set_offset   -> setOffset
- set_indexlength  ->  setIndexLength
- set_datalength  ->  setDataLength
- set_footerlength  ->  setFooterLength
- set_groupsize  ->  setGroupSize